oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexHelper.java

```diff
@@ -276,6 +276,11 @@ private static void mapIndexRules(ElasticIndexDefinition indexDefinition, XConte
             mappingBuilder.endObject();
         }
 
+        mappingBuilder.startObject(ElasticIndexDefinition.DYNAMIC_BOOST_TAGS)
+                .field("type", "text")
+                .field("analyzer", "oak_analyzer")
+                .endObject();
+
```

Comment:
I would change this. We currently support one or more dynamicBoost properties. Setting this to a fixed field won't work in case more than one field is configured. The same mapping can be moved in the dynamicBoost for-loop after line 276. I would also rename the field name using the `:fulltext` suffix:

```java
 mappingBuilder.startObject(pd.nodeName + FieldNames.FULLTEXT)
    .field("type", "text")
    .field("analyzer", "oak_analyzer")
    .endObject();
```

This would require changes in `ElasticDocument` and `ElasticRequestHandler`

Reply:
@fabriziofortino Thanks, I just refined the PR with another commit over the review. Can you review again to see if all meet what you suggested here?

Reply:
@FrancoisZhang I don't think this works. The mapping field in the for-loop overwrites itself since the name is always the same. The name has to be composed (see my suggestion above). I would create it with `pd.nodeName + FieldNames.FULLTEXT` (you can get rid of DYNAMIC_BOOST_TAGS). This would require some changes in `ElasticRequestHandler` since the fields to query need to be re-computed.

Reply:
@fabriziofortino just pushed another commit which change it back to a fixed field for storing all terms for multiple boost fields. as the query will do cross fields query. so either single field or multiple fields should works here. thanks

Reply:
@fabriziofortino OK, just pushed more tests to cover multiple dynamic boost cases. It's ready for merging if you also approve it, thanks!